src/bridge.py

```diff
@@ -1,184 +1,87 @@
+    pub_t8.publish(data.HEAVE_STERN_PORT)
+
+
+def cb_sim_pose(data):
     clarke_poses = data.poses[0]
     clarke_position = clarke_poses.position
     clarke_orientation = clarke_poses.orientation
-    pub_state_x.publish(clarke_position.x + 3)
-    pub_state_y.publish(clarke_position.y)
-    pub_state_z.publish(clarke_position.z)
-    
-    pose = Pose()
-    pose.position.x = clarke_position.x + 3
-    pose.position.y = clarke_position.y
-    pose.position.z = clarke_position.z
-    pose.orientation = clarke_orientation
-    pub_pose.publish(pose)
-
-    q = np.array([clarke_orientation.x, clarke_orientation.y, clarke_orientation.z, clarke_orientation.w])
-    conversion = transformations.euler_from_quaternion(q, 'rxyz')
-    theta_x = conversion[0]
-    theta_y = conversion[1]
-    theta_z = conversion[2]
-
-    angles = np.array([theta_x, theta_y, theta_z])*DEG_PER_RAD
-
-    for i in range(3):
-            if angles[i] - euler[i] > ANGLE_CHANGE_TOL:
-                euler[i] = angles[i] - 360
-            elif euler[i] - angles[i] > ANGLE_CHANGE_TOL:
-                euler[i] = angles[i] + 360
-            else:
-                euler[i] = angles[i]
-
-    pub_state_theta_x.publish(euler[0])
-    pub_state_theta_y.publish(euler[1])
-    pub_state_theta_z.publish(euler[2])
     
-def callback_imu(data):
+    dr_msg = DeadReckonReport()
+    dr_msg.x = clarke_position.x + 3 # TODO - sketch
+    dr_msg.y = clarke_position.y
+    dr_msg.z = clarke_position.z
+
+    depth = clarke_position.z # TODO - check with reference to which datum
+
+    # TODO - specify roll/pitch/yaw for dvl
+    pub_dvl_deadreckon.publish(dr_msg)
+    pub_depth_sensor.publish(depth)
+
+
+def cb_sim_imu(data):
+    # TODO - attach imu link to preserve orientation
     gyro = Vector3(-data.angular_velocity.z, -data.angular_velocity.y, -data.angular_velocity.x)
-    angular_velocity_pub.publish(gyro)
+    pub_imu_angular_vel.publish(gyro)
     
-    imu_ekf_quat_msg = SbgEkfQuat()
-    quat = Quaternion(-data.orientation.z, -data.orientation.y, -data.orientation.x, data.orientation.w)
-    imu_ekf_quat_msg.quaternion = quat
-    imu_quat_pub.publish(imu_ekf_quat_msg)
+    sbg_quat = SbgEkfQuat()
+    quat = Quaternion(-data.orientation.z, -data.orientation.y, -data.orientation.x, data.orientation.w) # TODO - incorrect?
+    sbg_quat.quaternion = quat
+    pub_imu_quat.publish(sbg_quat)
     
 
 if __name__ == '__main__':
 
     rospy.init_node('bridge')
 
-    pub_state_x = rospy.Publisher('state_x', Float64, queue_size=1)
-    pub_state_y = rospy.Publisher('state_y', Float64, queue_size=1)
-    pub_state_z = rospy.Publisher('state_z', Float64, queue_size=1)
-    pub_state_theta_x = rospy.Publisher('state_theta_x', Float64, queue_size=1)
-    pub_state_theta_y = rospy.Publisher('state_theta_y', Float64, queue_size=1)
-    pub_state_theta_z = rospy.Publisher('state_theta_z', Float64, queue_size=1)
-    pub_pose = rospy.Publisher('pose', Pose, queue_size=1)
```

Comment:
don't we need to have a pose publisher? or does dead_reckon_report take care of that?

Reply:
state_aggregator should take care of it as long as we are feeding data into the depth, imu and dvl topics it is subscribed to